### DISCUSSION TITLE
Nested NER

### DISCUSSION THREAD

**Discussion by SoimulPatriei** [explosion/spaCy#8496] [Help: Coding & Implementations] [answered]
Hi,
Is there any support for performing a Nested NER by fine-tuning BERT? Any snippet of code or advice using spacy to perform nested NER would be very much appreciated. 

Thanks!
Eduard

**Comment by svlandeg** [comment, marked as answer]
Hi!

The "traditional" `ner` task in spaCy doesn't support nested or overlapping spans, as each token can only be part of 1 named entity at most. However, we've been working on a new component called `spancat`, which does allow nested and overlapping spans: https://github.com/explosion/spaCy/pull/6747. We'll be releasing it as an experimental functionality from 3.1 onwards, which will be available soon. You could already try it out if you feel like installing the `master` branch from source. 

We're currently doing many experiments with this `spancat` functionality, including making sure it works with a `transformer` instead of a normal `tok2vec` embedding. For now I think it'll be more stable on CPU & with a "simple" Tok2Vec layer.

If you do decide to check it out - we'd love to hear your feedback on it!

**Reply by SoimulPatriei** [reply]
Very nice to hear you are considering this functionality. I would be happy to check the component after I will be back from vacation in August. Meanwhile, I will implement a baseline for nested NER that will fine-tune two BERT models with non-overlapping entities from the nested ones to maximize the coverage of entities. My raw estimation is that I will "cover" around 90% of training entities in this way.
